xwiki-platform-core/xwiki-platform-eventstream/src/main/java/org/xwiki/eventstream/internal/DefaultRecordableEventDescriptorContainer.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.eventstream.internal;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.eventstream.RecordableEventDescriptor;
+import org.xwiki.eventstream.RecordableEventDescriptorContainer;
+
+/**
+ * Default implementation of {@link RecordableEventDescriptorContainer}.
+ *
+ * @version $Id$
+ * @since 9.5RC1
+ */
+@Component
+@Singleton
+public class DefaultRecordableEventDescriptorContainer implements RecordableEventDescriptorContainer
+{
+    private List<RecordableEventDescriptor> recordableEventDescriptorList = new ArrayList<>();
+
+    @Override
+    public void addRecordableEventDescriptor(RecordableEventDescriptor recordableEventDescriptor)
+    {
+        this.recordableEventDescriptorList.add(recordableEventDescriptor);
+    }
+
+    @Override
+    public void deleteRecordableEventDescriptor(RecordableEventDescriptor recordableEventDescriptor)
+    {
+        this.recordableEventDescriptorList.remove(recordableEventDescriptor);
+    }
+
+    @Override
+    public List<RecordableEventDescriptor> getAllRecordableEventDescriptor() {
+        return this.recordableEventDescriptorList;
```

Comment:
@gdelhumeau seems to suggest that all this is supposed to be internal so It's less of an issue, otherwise returning directly a modifiable list here would be quite dangerous.